src/Application/Regulation/Query/Visa/GetVisasAndReasonsByRegulationOrderQueryHandler.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Application\Regulation\Query\Visa;
+
+use App\Application\Regulation\View\VisasAndReasonsView;
+use App\Domain\Regulation\Repository\RegulationOrderRepositoryInterface;
+
+final class GetVisasAndReasonsByRegulationOrderQueryHandler
+{
+    public function __construct(
+        private RegulationOrderRepositoryInterface $regulationOrderRepository,
+    ) {
+    }
+
+    public function __invoke(GetVisasAndReasonsByRegulationOrderQuery $query): VisasAndReasonsView
+    {
+        $visasAndReasons = $this->regulationOrderRepository->findVisasAndReasonsByRegulationOrderUuid($query->regulationOrderUuid);
```

Comment:
Traitement du cas où l'arrêté recherché n'existe pas

```suggestion
        $visasAndReasons = $this->regulationOrderRepository->findVisasAndReasonsByRegulationOrderUuid($query->regulationOrderUuid);
        
        if (!$visasAndReasons) {
           return new VisasAndReasonsView();
       }
```